src/sonic-bgpcfgd/tests/test_ipv6_nexthop_global.py

```diff
@@ -79,28 +79,34 @@ def extract_rm_from_peer_group(path, peer_group_name):
 def check_routemap_in_file(filename, route_map_name):
     route_map_re = re.compile(r'^route-map\s+%s\s+permit\s+(\d+)' % route_map_name)
     set_re = re.compile(r'set ipv6 next-hop prefer-global')
+    next_re = re.compile(r'on-match next')
     with open(filename) as fp:
         lines = [line.strip() for line in fp if not line.strip().startswith('!') and line.strip() != '']
     found_entry = False
     found_seq_no = None
     route_map_entries = {}
     for line in lines:
         if found_entry:
-            route_map_entries[found_seq_no] = set_re.match(line) is not None
-            found_entry = False
-            found_seq_no = None
+            if set_re.match(line):
+                route_map_entries[found_seq_no][0] = True
+            elif next_re.match(line):
+                route_map_entries[found_seq_no][1] = True
+            else:
+                found_entry = False
+                found_seq_no = None
         if route_map_re.match(line):
             found_seq_no = None
             seq_n_txt = route_map_re.match(line).group(1)
             assert seq_n_txt.isdigit(), "wrong sequence number for line '%s'" % line
             found_seq_no = int(seq_n_txt)
             assert found_seq_no not in route_map_entries, "Route-map has duplicate entries: %s - %d" % (route_map_name, found_seq_no)
             found_entry = True
+            route_map_entries[found_seq_no] = [False, False]
     results = [route_map_entries[seq] for seq in sorted(route_map_entries.keys())]
-    if (len(results)):
-        err_msg = "route-map %s doesn't have mandatory permit entry for 'set ipv6 next-hop prefer-global" % route_map_name
-        assert results[0], err_msg
-    return len(results) > 0
+    err_msg = "route-map %s doesn't have mandatory permit entry for 'set ipv6 next-hop prefer-global" % route_map_name
+    assert len(results), err_msg
```

Comment:
the err_msg here does not match exactly the assertion.

Reply:
the results arrays contains a list of pairs.
Each pair corresponds to the permit entry in ipv6 route-map
First permit entry in ipv6 route-map should contains 'set ipv6 ....' (first element in the result element) and 'on-match next' (second element in result element). So if the result array empty - we don't have any permit elelement in ipv6 route-map